pkgs/applications/misc/blender/default.nix

```diff
@@ -1,243 +1,51 @@
-{ config, stdenv, lib, fetchurl, fetchzip, boost, cmake, ffmpeg, gettext, glew
-, ilmbase, libepoxy, libXi, libX11, libXext, libXrender
-, libjpeg, libpng, libsamplerate, libsndfile
-, libtiff, libwebp, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio, openjpeg, python310Packages
-, openvdb, libXxf86vm, tbb, alembic
-, zlib, zstd, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
-, jackaudioSupport ? false, libjack2
-, cudaSupport ? config.cudaSupport, cudaPackages ? { }
-, hipSupport ? false, rocmPackages # comes with a significantly larger closure size
-, colladaSupport ? true, opencollada
-, spaceNavSupport ? stdenv.isLinux, libspnav
-, makeWrapper
-, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
-, waylandSupport ? stdenv.isLinux, pkg-config, wayland, wayland-protocols, libffi, libdecor, libxkbcommon, dbus
-, potrace
-, openxr-loader
-, embree, gmp, libharu
-, openpgl
-, mesa
-, runCommand
+{ lib
+, stdenv
+, llvmPackages_10
+, darwin
+, callPackage
+, python310
```

Comment:
> When the user wants to override python, should they `(blenderVersions.override { python = ...; }.stable` or `blender.override { python = ...; }`?

@SomeoneSerge Since `blender` is just an alias for `blenderVersions.stable`, `blender.override { python = ...; }` is the same as `blenderVersions.stable.override { python = ...; }`

Reply:
Actually, I'm not even sure you can override python

Reply:
@SomeoneSerge I don't think you can even override python for this one

Reply:
Well, my thinking was: ok, both `default.nix` and `generic.nix` take `python` as an argument (now `python310Packages` and `pythonPackages` resp.), so I can override either of them if I want to patch python. I probably want to do the most local override I can, i.e. `blender.override`. But is it safe, and how do I know whether `blender` consumes something else from `blenderVersions` that might have remembered the old python? (the answer is to read the code obviously)

Although now both should work, so maybe anything goes

Reply:
(moving the conversation back from [matrix](https://matrix.to/#/%23dev%3Anixos.org/%24N4UWVWsm6lh72clnVbJ-ongpUtbs77n8owzhty1WkM8?via=someonex.net&via=matrix.org&via=snug.si&via=turbo.ooo)) What you could do is remove any mentions of python from `default.nix`, and take both `python310Packages` and `pythonPackages ? python310Packages` in `generic.nix`

But I'm not saying you should. This is susceptible to yak shaving

Reply:
> Well, my thinking was: ok, both `default.nix` and `generic.nix` take `python` as an argument (now `python310Packages` and `pythonPackages` resp.), so I can override either of them if I want to patch python. I probably want to do the most local override I can, i.e. `blender.override`. But is it safe, and how do I know whether `blender` consumes something else from `blenderVersions` that might have remembered the old python? (the answer is to read the code obviously)
> 
> Although now both should work, so maybe anything goes

(didn't notice sorry)

All the user needs to worry about is to override the package itself, not `blenderVersions`, in the same vein as `minecraftServers`, where it uses `javaPackages` as inputs grabs specific versions of JRE for its `derivation.nix`

I don't think there's anything I could do.

Reply:
@SomeoneSerge ie unless `generic.nix` should have dependency logic (which version of dependency for which), I don't think I can avoid this